src/collectors/elasticsearch/elasticsearch.py

```diff
@@ -274,11 +274,15 @@ def collect_instance(self, alias, scheme, host, port):
 
         metrics['indices.datastore.size'] = indices['store']['size_in_bytes']
 
-        transport = data['transport']
-        metrics['transport.rx.count'] = transport['rx_count']
-        metrics['transport.rx.size'] = transport['rx_size_in_bytes']
-        metrics['transport.tx.count'] = transport['tx_count']
-        metrics['transport.tx.size'] = transport['tx_size_in_bytes']
+        try:
+            transport = data['transport']
```

Comment:
I assume this is what's null, correct? Instead of a try/catch what about an existence check?

Reply:
Correct. I like it.

Reply:
People sat try-except is faster than if :)   (...until it met with an exception)
https://www.geeksforgeeks.org/try-except-vs-if-in-python/

Reply:
I'm not worried about speed. I recommended the change because the `if` truly checks what you're looking for: missing transport metrics.

Try/catch blocks can obfuscate other exceptions and classify them as the one thing you were looking for. I know this case is pretty cut and dry, this is more about building a habit.

Reply:
Interesting. If you look in other places throughout the collector, namely line 285, you'll see it seems the original author preferred if. I'm fine either way!